view/src/Result.js

```diff
@@ -0,0 +1,218 @@
+// Representation of found differences of semantic comparision
+// provided in the form of YAML.
+// Author: Lukas Petr
+
+function compareFunctionNames(nameA, nameB) {
+  return nameA.toLowerCase().localeCompare(nameB.toLowerCase());
+}
+/**
+ * Result of diffkemp compare obtained from YAML file.
+ */
+export default class Result {
+  /**
+   * Map
+   * - keys: names of compared functions
+   * - values: Map
+   *   - keys: names of differing functions for the compared function
+   *   - values: diff (part of YAML) for the compared and differing function
+   * Note: The names of functions are inserted in sorted order.
+   */
+  #allCompFuns;
+
+  /**
+   * Map
+   * - keys: names of differing functions
+   * - values: sorted array of names of compared functions which differs in the differing function
+   * Note: The names of functions are inserted in sorted order.
+   */
+  #allDiffFuns;
+
+  /**
+   * Create Result from YAML file.
+   * @param {Object} yaml YAML file describing result of diffkemp compare.
+   */
+  constructor(yaml) {
+    Object.assign(this, yaml);
+    this.#sortComparedFunctionNames();
+    this.#sortDifferingFunctionNames();
+    this.#createAllCompAndDiffFuns();
```

Comment:
Copying `yaml` to `self` doesn't make much sense. Also, I'm not sure if putting the sorting code info functions helps here.

I'd suggest something like this:
```suggestion
    // Sort result by compared function name
    const results = [...yaml.results].sort((resultA, resultB) => (
      compareFunctionNames(resultA.function, resultB.function)
    ));
    // Sort differing functions of each compared function
    results.forEach((result) => {
      result.diffs.sort((diffA, diffB) => compareFunctionNames(diffA.function, diffB.function));
    });
    this.#createAllCompAndDiffFuns(results);
```
The spread syntax (`[...]`) should create a *shallow* copy which is then sorted in-place without modifying the original array. This seems like the most efficient approach to me if we don't want to sort `yaml.results` in-place.

Reply:
Thanks. I left it to be sorted in-place, I originally used `assign` (copying `yaml` to `self`) for another reason -- I wanted the object to contain all fields of the YAML -- now I solved it differently.